Source/Core/Core/HW/GPFifo.cpp

```diff
@@ -56,9 +56,23 @@ void Init()
   memset(s_gather_pipe, 0, sizeof(s_gather_pipe));
 }
 
-bool IsEmpty()
-{
-  return GetGatherPipeCount() == 0;
+bool IsBNE()
+{
+  // TODO: It's not clear exactly when the BNE (buffer not empty) bit is set.
+  // The PPC 750cl manual says in section 2.1.2.12 "Write Pipe Address Register (WPAR)" (page 78):
+  // "A mfspr WPAR is used to read the BNE bit to check for any outstanding data transfers."
+  // In section 9.4.2 "Write Gather Pipe Operation" (page 327) it says:
+  // "Software can check WPAR[BNE] to determine if the buffer is empty or not."
+  // On page 327, it also says "The only way for software to flush out a partially full 32 byte
+  // block is to fill up the block with dummy data,."
```

Comment:
Is this meant to have a comma and a period here?

Reply:
The manual itself is like that (and that's not the only typo it has). I chose to copy it exactly, because if the manual has a typo like that, it seems a bit more reasonable that it also is contradictory regarding the behavior.

Reply:
Okay, just making sure.